portable/GCC/ARM_CM4F/port.c

```diff
@@ -543,23 +534,45 @@ void xPortSysTickHandler( void )
 		to be unsuspended then abandon the low power entry. */
 		if( eTaskConfirmSleepModeStatus() == eAbortSleep )
 		{
-			/* Restart from whatever is left in the count register to complete
-			this tick period. */
-			portNVIC_SYSTICK_LOAD_REG = portNVIC_SYSTICK_CURRENT_VALUE_REG;
-
-			/* Restart SysTick. */
-			portNVIC_SYSTICK_CTRL_REG |= portNVIC_SYSTICK_ENABLE_BIT;
-
-			/* Reset the reload register to the value required for normal tick
-			periods. */
-			portNVIC_SYSTICK_LOAD_REG = ulTimerCountsForOneTick - 1UL;
-
 			/* Re-enable interrupts - see comments above the cpsid instruction()
 			above. */
 			__asm volatile( "cpsie i" ::: "memory" );
 		}
 		else
 		{
+			/* Stop the SysTick momentarily.  The time the SysTick is stopped for
+			is accounted for as best it can be, but using the tickless mode will
+			inevitably result in some tiny drift of the time maintained by the
+			kernel with respect to calendar time. */
+			portNVIC_SYSTICK_CTRL_REG = ( portNVIC_SYSTICK_CLK_BIT_SETTING | portNVIC_SYSTICK_INT_BIT );

```

Comment:
Note that portNVIC_SYSTICK_CLK_BIT_SETTING (with the _SETTING suffix) is configuration dependent and holds the correct value for the CLK bit (bit 2), whether set or clear.  So in reality the instruction as written really is just clearing the enable bit without any other side effects.  Just like the original code [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/4b353bfd7aa009d553f31b4e47dfb49a9f3c75b1/portable/GCC/ARM_CM4F/port.c#L610).  It's more efficient since it avoids have to do read-modify-write.

This method of stopping systick (a simple write instead of RMW) is required the second time we stop systick (below).  Because this method is already required in this function and is more efficient than RMW, I used it for the first stopping of systick here too.  Thoughts?